### PR TITLE
Fix c-ffi test printing

### PR DESCRIPTION
**Pull request by zachschuermann**
small fix since string isn't null terminated

### CODE REVIEW

**Review by nicklan — APPROVED** [review]
lgtm

File: ffi/cffi-test.c [hunk 1]
```diff
@@ -37,7 +42,7 @@ int main(int argc, char* argv[]) {
   const SnapshotHandle *snapshot_handle = snapshot_handle_res.ok;
 
   uint64_t v = version(snapshot_handle);
-  printf("version: %llu\n", v);
+  printf("version: %" PRIu64 "\n", v);
```
Comment (scovich) [inline ffi/cffi-test.c:45]: Could also potentially do:
```c++
    size_t v = version(snapshot_handle);
    printf("version: %zu\n", v);
```
(technically that's C99 not C++, but I'm pretty sure both clang and gcc support it out of the box as an extension)

File: ffi/cffi-test.c [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
+#include <inttypes.h>
 
 #include "delta_kernel_ffi.h"
 
 void visit_file(void *engine_context, struct KernelStringSlice file_name) {
-    printf("file: %s\n", file_name.ptr);
+    printf("file: ");
+    for (int i = 0; i < file_name.len; i++) {
+        printf("%c", file_name.ptr[i]);
+    }
```
[the remark this answers is not in the document]
Comment (zachschuermann) [reply ffi/cffi-test.c:11]: hm didn't seem to like that
```
cffi-test.c:8:21: warning: field precision should have type 'int', but argument has type 'uintptr_t' (aka 'unsigned long') [-Wformat]
    printf("file: %.*s\n", file_name.len, file_name.ptr);
                  ~~^~     ~~~~~~~~~~~~~
1 warning generated.
```

File: ffi/cffi-test.c (hunk 1 above)
Comment (zachschuermann) [reply ffi/cffi-test.c:45]: I think this worked - I'm on clang FWIW

Comment (zachschuermann) [reply ffi/cffi-test.c:45]: @nicklan has been our gcc tester

File: ffi/cffi-test.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (scovich) [reply ffi/cffi-test.c:11]: (that technically won't print any NUL characters the string might contain, but that's probably fine)

File: ffi/cffi-test.c (hunk 1 above)
Comment (hntd187) [reply ffi/cffi-test.c:45]: Everything breaks on MSVC :-)

**Review by scovich — APPROVED** [review]

**Pull request merged**